src/request_template.rs

```diff
@@ -82,6 +119,23 @@ impl RequestTemplate {
     Ok(req)
   }
 
+  // /// A high-performance way to reliably create a request
+  pub fn to_request2<C: PathString + HasHeaders>(&self, ctx: &C) -> anyhow::Result<reqwest::Request> {
```

Comment:
RequestTemplate should keep a `Option<Request>` so that the request doesn't need to be created again and again in scenarios where the mustache templates are all literals and don't need to be evaluated again and again. It will still need to be updated based on the ctx, but that would still be an optimization.

Reply:
@tusharmath  How common of a use case is it to have all of the mustache objects be literals?  Also, does mustache currently have a function to tell if a mustache object is composed only of literals and constants?

Reply:
@tusharmath  Actually mustache templates shouldn't interfere with caching. Either the request_to function should just resolve all the mustache templates at the beginning and then check if the inputs have already been cached or to_request shouldn't know it's working on mustache templates in the first place. In order to allow to_request to work on mustache templates without knowing about it you would need to create a mustache iterator and a mustache map. If you had a mustache iterator that could just iterate through a list of mustache templates you could use mustache templates on any function that takes an iterator as input.

Reply:
> How common of a use case is it to have all of the mustache objects be literals? 

Common for resolvers that are provided at the root of the Query.


> Also, does mustache currently have a function to tell if a mustache object is composed only of literals and constants?

Mustache could implement a method called `is_literal()` which could iterate over the template and see if everything is a literal. 

To your second point, I agree, and this is how we can go about it — 
- Let RequestTemplate (RT) have a `Option<Request>` inside it.
- While creating a RT from an Endpoint, we will check if `is_const()` is true for the `Endpoint` We need `is_const` on `Endpoint` that checks if every mustache template is a literal in the Endpoint.
- If `is_literal` is true, we create a Request AOT and save it inside the RT.
- While calling `to_request` we check if AOT `Request` exists, if it does, return a ref otherwise, create a new one and return. Don't update cache.


This way `to_request` doesn't care about `Mustache` and it's internals. It just performs an AOT on the Endpoint if we know that it's not going to change.

Reply:
@tusharmath  this has now been added